### PR TITLE
Add IsTextureFormatSupported and GetBestSampleCount functions

### PR DESCRIPTION
**Pull request by TheSpydog**
This PR adds two new functions to the SDL_GPU API, for querying texture format capabilities.

One of them is very simple -- it just gets the "best" supported multisample count that is <= a desired sample count. Just like in FNA3D.

The other function is more complicated. Not all formats are supported on all hardware, and even hardware that does support a format does not necessarily support it in all contexts. A particular format may be compatible with the COLOR_TARGET usage but only if the texture is 2D, not a cubemap. Another format may support all usages except compute. There are, sadly, many such cases. It gets complicated fast.

The `IsTextureFormatSupported` function aims to solve that problem. You can simply ask SDL_GPU, "Hey, is it ok to use format X with texture type Y for usage Z?" And you'll get an SDL_bool with the answer. The main use case I have in mind for this is depth-stencil format queries. The client can check if (e.g.) D24_S8 is supported, and if not, manually fall back to D32_S8. But you could also do this for other "risky" texture formats that may not be supported on all the target hardware. Maybe a certain machine doesn't support COMPUTE usage with a certain texture type, in which case you can perform your calculations CPU-side instead.

However, requiring manual fallback logic is a burden to put on the client, so for (relatively) trivial cases like depth-stencil, we could handle it automatically in each of the SDL_GPU backends.

However... I think we can do one better: SDL_gpu.c itself could perform the fallback logic. Before calling into the driver's CreateTexture function, we can sneak a quick peek at the supported usages/types for the TextureCreateInfo. If we're trying to create a texture with an unsupported depth format, we can swap it out in the create info there. By the time the backend gets the texture create info, all the data will be valid. (We could also sneak in some other validation here if we wanted.)

I have not implemented the above, because it does add some overhead for each texture create call and I wanted to run this idea past you all first. (Although realistically the overhead will probably be inconsequential compared to the cost of actually, you know, creating a texture.)

We may also want to add a similar function for checking buffer support, since buffers also have usage flags.

### CODE REVIEW

File: include/SDL3/SDL_gpu.h [hunk 1]
```diff
@@ -121,8 +121,9 @@ typedef enum SDL_GpuTextureFormat
 	SDL_GPU_TEXTUREFORMAT_R16G16B16A16_UINT,
 	/* Depth Formats */
 	SDL_GPU_TEXTUREFORMAT_D16_UNORM,
```
Comment (kg) [inline include/SDL3/SDL_gpu.h:123]: is d16 widely supported and it's just d16_s8 that isn't?

File: src/gpu/vulkan/SDL_gpu_vulkan.c [hunk 2]
```diff
@@ -213,10 +213,11 @@ static VkFormat RefreshToVK_SurfaceFormat[] =
     VK_FORMAT_R16_UINT,					/* R16_UINT */
     VK_FORMAT_R16G16_UINT,				/* R16G16_UINT */
     VK_FORMAT_R16G16B16A16_UINT,		/* R16G16B16A16_UINT */
-    VK_FORMAT_D16_UNORM,				/* D16_UNORM */
-    VK_FORMAT_D32_SFLOAT,				/* D32_SFLOAT */
-    VK_FORMAT_D16_UNORM_S8_UINT,		/* D16_UNORM_S8_UINT */
-    VK_FORMAT_D32_SFLOAT_S8_UINT		/* D32_SFLOAT_S8_UINT */
+	VK_FORMAT_D16_UNORM,				/* D16_UNORM */
```
Comment (kg) [inline src/gpu/vulkan/SDL_gpu_vulkan.c:216]: inconsistent indentation (in case the sdl people care about that)

File: include/SDL3/SDL_gpu.h (hunk 1 above)
Comment (TheSpydog) [reply include/SDL3/SDL_gpu.h:123]: Yeah, d16 is guaranteed to be supported by all APIs. For Vulkan it has 99.88% device coverage on gpuinfo, while d16_s8 has 22%.

File: src/gpu/vulkan/SDL_gpu_vulkan.c (hunk 2 above)
Comment (TheSpydog) [reply src/gpu/vulkan/SDL_gpu_vulkan.c:216]: Thanks, will address in the next push. That said, the formatting is generally a mess in these files. I think the plan is to clean everything up formally in one big pass once the dust has settled a bit.

**Review by flibitijibibo — APPROVED** [review]
lgtm - if we're worried about changing formats behind the application's back we can maybe take a note from XNA and change the format parameter to "preferredFormat" and document why the resulting internal format may be different. (This would also make sense for swizzled color formats, now that I think about it)

**Pull request merged**